### PR TITLE
Home and OCP Page

### PR DESCRIPTION
**Pull request by MVarshini**
## Type of change

- [ ] Refactor
- [X] New feature
- [ ] Bug fix
- [ ] Optimization
- [ ] Documentation Update

## Description

This PR has the Home and CPT page Tables with Filters, sorting, dynamic column selection and pagination 

## Related Tickets & Documents

- Related Issue #
- Closes #

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please describe the System Under Test.
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

### CODE REVIEW

File: local-compose.sh [hunk 1]
```diff
@@ -11,9 +11,11 @@ CPT_CONFIG=${CPT_CONFIG:-"$PWD/backend/ocpperf.toml"}
 podman rm -f front back
 
 podman build -f backend/backend.containerfile --tag backend
-podman build -f frontend/frontend.containerfile --tag frontend
+# podman build -f frontend/frontend.containerfile --tag frontend
```
Comment (vishnuchalla) [inline local-compose.sh:14]: This needs to be un-commented i believe

**Comment by vishnuchalla** [comment]
@MVarshini Can you also get our recent `Telco` integration added here as well?
Screenshot from 2024-07-02 14-59-51

**Review by chentex — CHANGES_REQUESTED** [review]
Great job @MVarshini!!
There are a couple of problems I found:

- Seems that the Sorting on the columns is not working on any page
- The values on the filters seem to repeat, I see in some `FALSE` and `false` both filter the same

And a couple of requests:

- Can we make filters multi-select?
- Can we recalculate the values on the filters based on the selected filter values?

**Comment by MVarshini** [comment]
@chentex Thanks for the suggestions. I'll address the mentioned issues.

**Comment by MVarshini** [comment]
@chentex Thanks for your valuable suggestions. 

The Sorting and repeating values issues are resolved. 

And, the request for multi-select in filter and updating the filter values based on previous selection will be addressed in a separate PR and is tracked in [PANDA-446](https://issues.redhat.com/browse/PANDA-446)

File: local-compose.sh (hunk 1 above)
Comment (MVarshini) [reply local-compose.sh]: Done

**Comment by vishnuchalla** [comment]
@MVarshini `telco` specific tab doesn't seem to be working. Looks like a redirection problem. It seems to appear well in the homepage but not in the product specific page. Can you cross-check please?

File: frontend/src/App.jsx [hunk 2]
```diff
@@ -20,7 +21,7 @@ function App() {
           <Route element={<MainLayout />}>
             <Route index element={<Home />} />
             <Route path={APP_ROUTES.HOME} element={<Home />} />
-            <Route path={APP_ROUTES.OCP} element={<Home />} />
+            <Route path={APP_ROUTES.OCP} element={<OCP />} />
             <Route path={APP_ROUTES.QUAY} element={<Home />} />
```
Comment (vishnuchalla) [inline frontend/src/App.jsx:25]: Wrong mapping here. Also telco doesn't seem to be added here.

Comment (MVarshini) [reply frontend/src/App.jsx]: Updated!

Comment (vishnuchalla) [reply frontend/src/App.jsx]: Quay is mapping to `Home`, instead it should be `Quay` itself. And still there is no mapping for `Telco`,

**Review by vishnuchalla — APPROVED** [review]
@MVarshini assuming that we have plans to write up frontend reducers and actions for Quay and Telco in the subsequent PRs, these changes look good to me.

**Comment by chentex** [comment]
As we talked with @MVarshini `telco` and `quay` are going to be worked on another PR

**Pull request merged**